assets/js/blocks/checkout/block.tsx

```diff
@@ -84,22 +84,29 @@ const Checkout = ( {
 		return <CheckoutOrderError />;
 	}
 
+	/**
+	 * Show the login prompt if the user is not logged in
+	 * AND no guest checkout is allowed
+	 * AND no account creation is allowed..
+	 */
 	if (
 		isLoginRequired( customerId ) &&
-		getSetting( 'checkoutAllowsSignup', false )
+		! getSetting( 'checkoutAllowsSignup', false )
 	) {
-		<LoginPrompt />;
+		return <LoginPrompt />;
```

Comment:
The <LoginPrompt> component returns `You must be logged in to checkout.
[Click here to log in.](https://woo.local/my-account/?redirect_to=https%3A%2F%2Fwoo.local%2Fcheckout-2%2F)`. There is no space between the two sentences, could you add a non breaking space there as part of this PR just to tidy it up a bit?

Reply:
Good catch, @alexflorisca. I noticed this initially, but then forgot to address this part. It took me a while to figure out what happened in this case. Initially, the text `You must be logged in to checkout. ` had a trailing space as part of the translation string:

```tsx
{ __(
    'You must be logged in to checkout. ',
    'woo-gutenberg-products-block'
) }
```

Technically, this space should have been visible. It turned out that we're wrapping the Checkout block with the CSS class `.wc-block-components-sidebar-layout` which is defined on https://github.com/woocommerce/woocommerce-blocks/blob/c69f13d7e6ab1b1ef7ba528e7b7015d6aacea179/assets/js/base/components/sidebar-layout/style.scss#L1-L5

As the out box is using `display: flex;`, the space behind the text gets trimmed. Removing the CSS class `.wc-block-components-sidebar-layout` would not be an option, as this would affect the styles of the Checkout block. 

I addressed this issue by wrapping the login text and login link with the CSS class `.wc-block-must-login-prompt`. In https://github.com/woocommerce/woocommerce-blocks/blob/0fc0177aa1393fbb7bf2491f978b06d7d2c56ae9/assets/js/base/components/sidebar-layout/style.scss#L7-L9, I then set the display setting to `display: block;`. This way, the trailing space became visible. I went a step further and removed the space from the translation changed

```tsx
{ __(
	'You must be logged in to checkout. ',
	'woo-gutenberg-products-block'
) }
<a href={ LOGIN_TO_CHECKOUT_URL }>
	{ __(
		'Click here to log in.',
		'woo-gutenberg-products-block'
	) }
</a>
```

to 

```tsx
{ __(
	'You must be logged in to checkout.',
	'woo-gutenberg-products-block'
) }{ ' ' }
<a href={ LOGIN_TO_CHECKOUT_URL }>
	{ __(
		'Click here to log in.',
		'woo-gutenberg-products-block'
	) }
</a>
```

The reason I'm doing this, is that a translator might not be aware of the purpose of the trailing space and might not add that to a translation. A translation would then end up without the space between the login text and the login link.

Reply:
Thanks Niels, nice work. I don't think there's any need for the extra `wc-block-must-login-prompt` class. I've just removed it and the text still renders just fine with the space outside the translated text

Reply:
As discussed in Slack, we'll leave this part, as we're seeing different results in this case.